service/src/main/java/com/commercetools/util/HttpRequestUtil.java

```diff
@@ -79,10 +86,19 @@
                     .setConnectTimeout(REQUEST_TIMEOUT)
                     .build())
             .setRetryHandler(HTTP_REQUEST_RETRY_ON_SOCKET_TIMEOUT)
+            .setConnectionManager(buildDefaultConnectionManager())
             .build();
 
     private static final BasicResponseHandler BASIC_RESPONSE_HANDLER = new BasicResponseHandler();
 
+    private static PoolingHttpClientConnectionManager buildDefaultConnectionManager() {
```

Comment:
I would add a simple unit test to assert the default connection manager with such behaviour/values. To make sure that the that default connection manager always has such behaviour/values, so that no dev in the future changes the behaviour without knowing the side effects, if there are..

Reply:
I'd also love to do this, but created/returned `CloseableHttpClient` exposes only `execute()` and `close()` methods. The _pool_ and _timeout_ properties are hidden very deep in the implementation.

The only thing i was able to test are short/long timeout responses in the `functionaltests`. 
I'd also like to test pool request size, but for this i have to make a stand alone test, when all funtionaltests are run in parallel doing tens of requests per second

Reply:
ok, why not make the `buildDefaultConnectionManager` have `package-private` access?

Reply:
does it really have sense to test static stateless method with explicit logic? It is just a dumb   3-lines factory method with straightforward logic and without any branching, looping, extending or inheriting. Actually the only reason this method exists as a separate method is that `HttpClientConnectionManager` doesn't have a builder (like `RequestConfig` which is used in `setDefaultRequestConfig()` above). Hence i carried out `new->set->set` to separate method. Otherwise i would looks same as in `setDefaultRequestConfig()` or `DefaultHttpRequestRetryHandler`.

Reply:
My only suggestion was to have the test as a safeguard against any side effects if someone changes the code in the future. Not to test the actual complexity logic of this method.

Reply:
22fa8000 and be happy :)